devlib/target.py

```diff
@@ -62,10 +62,11 @@ def is_connected(self):
         return self.conn is not None
 
     @property
-    @memoized
```

Comment:
There should not be a setter -- as far as the user code is concerned, this property is read-only. Internally, the `self._connected_as_root` value should be updated directly, rather than via the property.